content/en/blog/_posts/2022-11-XX-runtime-observability/index.md

```diff
@@ -0,0 +1,275 @@
+running as well. There are multiple ways to run OTLP as [described in the
+docs][collector], but it's also possible to `kubectl proxy` into an existing
+instance running within Kubernetes.
+
+[collector]: https://opentelemetry.io/docs/collector/getting-started
+
+If everything is setup, then the collector should log that there are incoming
+traces:
+
+```
+ScopeSpans #0
+ScopeSpans SchemaURL:
+InstrumentationScope go.opentelemetry.io/otel/sdk/tracer
+Span #0
+    Trace ID       : 71896e69f7d337730dfedb6356e74f01
+    Parent ID      : a2a7714534c017e6
+    ID             : 1d27dbaf38b9da8b
+    Name           : github.com/cri-o/cri-o/server.(*Server).filterSandboxList
+    Kind           : SPAN_KIND_INTERNAL
+    Start time     : 2022-11-15 09:50:20.060325562 +0000 UTC
+    End time       : 2022-11-15 09:50:20.060326291 +0000 UTC
+    Status code    : STATUS_CODE_UNSET
+    Status message :
+Span #1
+    Trace ID       : 71896e69f7d337730dfedb6356e74f01
+    Parent ID      : a837a005d4389579
+    ID             : a2a7714534c017e6
+    Name           : github.com/cri-o/cri-o/server.(*Server).ListPodSandbox
+    Kind           : SPAN_KIND_INTERNAL
+    Start time     : 2022-11-15 09:50:20.060321973 +0000 UTC
+    End time       : 2022-11-15 09:50:20.060330602 +0000 UTC
+    Status code    : STATUS_CODE_UNSET
+    Status message :
+Span #2
+    Trace ID       : fae6742709d51a9b6606b6cb9f381b96
+    Parent ID      : 3755d12b32610516
+    ID             : 0492afd26519b4b0
+    Name           : github.com/cri-o/cri-o/server.(*Server).filterContainerList
+    Kind           : SPAN_KIND_INTERNAL
+    Start time     : 2022-11-15 09:50:20.0607746 +0000 UTC
+    End time       : 2022-11-15 09:50:20.060795505 +0000 UTC
+    Status code    : STATUS_CODE_UNSET
+    Status message :
+Events:
+SpanEvent #0
+     -> Name: log
+     -> Timestamp: 2022-11-15 09:50:20.060778668 +0000 UTC
+     -> DroppedAttributesCount: 0
+     -> Attributes::
+          -> id: Str(adf791e5-2eb8-4425-b092-f217923fef93)
+          -> log.message: Str(No filters were applied, returning full container list)
+          -> log.severity: Str(DEBUG)
+          -> name: Str(/runtime.v1.RuntimeService/ListContainers)
+```
+
+We can see that the spans have a trace ID and probably a parent attached. Events
+like logs are part of the output as well. In the above case the kubelet is
+periodically triggering a `ListPodSandbox` RPC to CRI-O caused by the Pod
+Lifecycle Event Generator (PLEG). Displaying those traces can be done via the
+[Jaeger][jaeger] project. When running the tracing stack locally, then a Jaeger
+instance should be exposed on `http://localhost:16686` per default.
+
+[jaeger]: https://www.jaegertracing.io/
+
+The `ListPodSandbox` requests are directly visible within the Jaeger UI:
+
+![ListPodSandbox RPC in the Jaeger UI](list_pod_sandbox.png)
+
+That's not too exciting, so let's run a workload directly via `kubectl`:
+
+```shell
```

Comment:
Avoid marking things as shell that aren't valid shell scripts. We're not able to predict how future highlighters will format invalid shell fragments, and it could end up looking weird.

Reply:
Alright, done